src/hotspot/share/prims/jvm.cpp

```diff
@@ -1326,10 +1326,13 @@ JVM_ENTRY(jclass, JVM_GetDeclaringClass(JNIEnv *env, jclass ofClass))
   if (!klass->is_instance_klass()) {
     return nullptr;
   }
+  InstanceKlass* k = InstanceKlass::cast(klass);
+
+  // Ensure class is linked
+  k->link_class(CHECK_NULL);
 
   bool inner_is_member = false;
-  Klass* outer_klass
-    = InstanceKlass::cast(klass)->compute_enclosing_class(&inner_is_member, CHECK_NULL);
+  Klass* outer_klass = k->compute_enclosing_class(&inner_is_member, CHECK_NULL);
```

Comment:
Why not put this change in compute_enclosing_class() instead?

Reply:
Various reflection methods have to work on a linked class, but that doesn't mean they all have to. I'm not convinced the true problem has been identified here yet.